help/blocks/search/search.js

```diff
@@ -1,22 +1,225 @@
-import { createTag, decorateIcons } from '../../scripts/lib-franklin.js';
+import { decorateIcons } from '../../scripts/lib-franklin.js';
+import { addPagingWidget } from '../../scripts/scripts.js';
 
-export default async function decorate(block) {
+/**
+ * Convert html in text form to document element
+ * @param {string} html the html to process
+ * @returns A document element
+ */
+function htmlToElement(html) {
+  const div = document.createElement('div');
+  div.innerHTML = html;
+  return div.firstElementChild;
+}
+
+function getSearchWidgetHTML(placeholders, initialVal, searchbox, formAction) {
+  const searchType = searchbox ? 'search' : 'text';
+  return `
+  <div class="search-outer">
+    <form method="get" class="search" action="${formAction}">
+      <label for="js-search-anywhere" class="search-label"></label>
+        <input type="${searchType}" name="q" value="${initialVal ?? ''}" id="js-search-anywhere" class="search-input"
+          placeholder="${placeholders.searchtext}" required="true" oninput="this.setCustomValidity('')"
+          oninvalid="this.setCustomValidity('${placeholders.emptysearchtext}')">
+        <button type="submit" tabindex="-1" aria-label="search" class="search-button">
+          <span class="icon icon-search"></span>
+        </button>
+      <div class="search-info">
+        <span class="icon icon-error"></span>
+        <span class="search-info-text">Input search</span>
+      </div>
+    </form>
+  </div>`;
+}
+
+function getSearchWidget(placeholders, initialVal, searchbox, formAction) {
+  const widget = getSearchWidgetHTML(placeholders, initialVal, searchbox, formAction);
+  return htmlToElement(widget);
+}
+const filterData = (indexItem, section, searchTerm) => {
+  const searchLower = searchTerm.toLowerCase().trim();
+  return (
+    indexItem.title.toLowerCase().includes(searchLower)
```

Comment:
Also, i see the number of filtered result is very different from what it is in the original website, anything we are missing?

Reply:
I discussed same with @asthabh23 and showed her that in multiple result pages that keyword does not exist.
Thus in the issue ,scope of search is mentioned that we will have to search the keyword in these  specific columns only.